packages/demo-nextjs/src/app/admin/(cms)/system/page.tsx

```diff
@@ -1,22 +1,107 @@
 "use client";
 
+import { KTCardBody } from "@/components/Metronic/helpers";
+import { KTTabs } from "@/components/Metronic/helpers/components/KTTabs";
+import { Loading } from "@/components/Metronic/helpers/components/Loading";
 import Root from "@/components/Metronic/layout/Root";
+import { Content } from "@/components/Metronic/layout/components/content";
+import { ToolbarWrapper } from "@/components/Metronic/layout/components/toolbar";
 import { PageTitle } from "@/components/Metronic/layout/core";
-import { CollectionsListWrapper } from "@/components/Metronic/modules/apps/collections/CollectionsList";
 import { Breadcrumbs } from "@/constants/Breadcrumbs";
-import React, { FC } from "react";
-
-const config = {
-  disabledFields: ["author", "status", "date", "selection"],
-  disableHeader: true,
-  disablePagination: true,
-};
+import { ProjectSettings } from "@/lib/db/types";
+import { projectId } from "@/utils/mock";
+import clsx from "clsx";
+import React, { FC, useCallback, useEffect, useMemo, useState } from "react";
+import { ProjectSettingsContext } from "./Context";
+import { ContextDataItem } from "./types";
+import { componentsTabs, tabs } from "./utils";
 
 const System: FC = () => {
+  const [data, setData] = useState<ProjectSettings | null>(null);
+  const [isFetching, setIsFetching] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setIsFetching(true);
+        const response = await fetch(`/api/projectSettings?id=${projectId}`);
+        const { data } = await response.json();
+        setData(data);
```

Comment:
use react-query for fetch-ing and all of state like loading, fettling, ettc use from react-query

Example:
![image](https://github.com/user-attachments/assets/71ec8459-c9b0-48fb-b65c-a47000978077)